google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -56,7 +56,13 @@ void ListObjects(google::cloud::storage::Client client, int& argc,
   //! [list objects] [START storage_list_files]
   namespace gcs = google::cloud::storage;
   [](gcs::Client client, std::string bucket_name) {
-    for (gcs::ObjectMetadata const& meta : client.ListObjects(bucket_name)) {
+    for (auto&& m : client.ListObjects(bucket_name)) {
+      if (not m.ok()) {
+        std::cerr << "Error reading object list for " << bucket_name
+                  << ", status=" << m.status();
+        return;
+      }
```

Comment:
FR: Add documentation that explains the different ways to handle errors in exceptions and no-exceptions. It will help point the user in the right direction from a lifecycle perspective when using this client.

Reply:
Acking in #1792.